### ISSUE TITLE
REPOS with mavenCentral and custom repos does not work

### ISSUE THREAD

**Issue by davsclaus** [jbangdev/jbang#713]
I have a little camel prototype in the works and wanted to launch it via jbang.

As it uses SNAPSHOT of Camel then I need to add ASF snapshot repo to the DEPS as done here
https://github.com/davsclaus/camel-bang/blob/main/CamelBang.java#L2

PS: In your docs you have `mavencentral` in the table, and in the code snippet its `mavenCentral`. So you are confused if the two spelling work the same?

When launching the app via

```
❯ jbang launch@davsclaus/camel-bang Foo.java
[jbang] Resolving dependencies...
[jbang]     Resolving org.apache.camel:camel-main:3.8.0-SNAPSHOT...[jbang] [ERROR] Could not resolve dependency
[jbang] Run with --verbose for more details
```

And then with verbose mode
```
❯ jbang --verbose launch@davsclaus/camel-bang Foo.java
[jbang] Reading aliases from /Users/davsclaus/.jbang/implicit-catalog.json
[jbang] Retrieved file from cache https://raw.githubusercontent.com/davsclaus/camel-bang/HEAD/jbang-catalog.json = /Users/davsclaus/.jbang/cache/urls/41350f57e8ba8970ad6b489f1c339acdda9912bd20d9a1bee353afc8df92643d/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/davsclaus/camel-bang/blob/HEAD/jbang-catalog.json
[jbang] Reading aliases from /Users/davsclaus/.jbang/cache/urls/41350f57e8ba8970ad6b489f1c339acdda9912bd20d9a1bee353afc8df92643d/jbang-catalog.json
[jbang] Reading aliases from /Users/davsclaus/.jbang/jbang-catalog.json
[jbang] Retrieved file from cache https://raw.githubusercontent.com/davsclaus/camel-bang/HEAD/jbang-catalog.json = /Users/davsclaus/.jbang/cache/urls/41350f57e8ba8970ad6b489f1c339acdda9912bd20d9a1bee353afc8df92643d/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/davsclaus/camel-bang/blob/HEAD/jbang-catalog.json
[jbang] Downloaded file https://raw.githubusercontent.com/davsclaus/camel-bang/HEAD/CamelBang.java
[jbang] System Java version detected as 11
[jbang] System Java version matches requested version 11
[jbang] Resolving dependencies...
[jbang]     Resolving org.apache.camel:camel-main:3.8.0-SNAPSHOT...[jbang] Deleting folder /Users/davsclaus/.jbang/cache/jars/CamelBang.java.887fff1e4dcc0e65777af70790fff6699a58b40bb481f487da4b17b106b673cf.jar.tmp
[jbang] [ERROR] Could not resolve dependency
dev.jbang.cli.ExitException: Could not resolve dependency
	at dev.jbang.dependencies.DependencyUtil.lambda$resolveDependenciesViaAether$3(DependencyUtil.java:196)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether(DependencyUtil.java:203)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:108)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:58)
	at dev.jbang.source.ScriptSource.resolveClassPath(ScriptSource.java:194)
	at dev.jbang.source.RunContext.resolveClassPath(RunContext.java:252)
	at dev.jbang.cli.BaseBuildCommand.buildJar(BaseBuildCommand.java:156)
	at dev.jbang.cli.BaseBuildCommand.build(BaseBuildCommand.java:130)
	at dev.jbang.cli.BaseBuildCommand.buildIfNeeded(BaseBuildCommand.java:93)
	at dev.jbang.cli.Run.prepareArtifacts(Run.java:84)
	at dev.jbang.cli.Run.doCall(Run.java:74)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:85)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:12)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
Caused by: org.jboss.shrinkwrap.resolver.api.NoResolvedResultException: Unable to collect/resolve dependency tree for a resolution due to: Failed to collect dependencies at org.apache.camel:camel-main:jar:3.8.0-SNAPSHOT, caused by: Could not find artifact org.apache:apache:pom:23 in asfsnap (http://repository.apache.org/snapshots/)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.wrapException(MavenWorkingSessionImpl.java:503)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.resolveDependencies(MavenWorkingSessionImpl.java:242)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.using(MavenStrategyStageBaseImpl.java:70)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.withTransitivity(MavenStrategyStageBaseImpl.java:52)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenStrategyStageBaseImpl.withTransitivity(MavenStrategyStageBaseImpl.java:39)
	at dev.jbang.dependencies.DependencyUtil.lambda$resolveDependenciesViaAether$3(DependencyUtil.java:189)
	... 27 more
Caused by: org.eclipse.aether.resolution.DependencyResolutionException: Failed to collect dependencies at org.apache.camel:camel-main:jar:3.8.0-SNAPSHOT
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:353)
	at org.jboss.shrinkwrap.resolver.impl.maven.bootstrap.MavenRepositorySystem.resolveDependencies(MavenRepositorySystem.java:121)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.resolveDependencies(MavenWorkingSessionImpl.java:239)
	... 31 more
Caused by: org.eclipse.aether.collection.DependencyCollectionException: Failed to collect dependencies at org.apache.camel:camel-main:jar:3.8.0-SNAPSHOT
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:291)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:309)
	... 33 more
Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for org.apache.camel:camel-main:jar:3.8.0-SNAPSHOT
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:301)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:171)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.resolveCachedArtifactDescriptor(DefaultDependencyCollector.java:541)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.getArtifactDescriptorResult(DefaultDependencyCollector.java:524)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:412)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:365)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.process(DefaultDependencyCollector.java:352)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:254)
	... 34 more
Caused by: org.apache.maven.model.resolution.UnresolvableModelException: Could not find artifact org.apache:apache:pom:23 in asfsnap (http://repository.apache.org/snapshots/)
	at org.apache.maven.repository.internal.DefaultModelResolver.resolveModel(DefaultModelResolver.java:176)
	at org.apache.maven.repository.internal.DefaultModelResolver.resolveModel(DefaultModelResolver.java:222)
	at org.apache.maven.model.building.DefaultModelBuilder.readParentExternally(DefaultModelBuilder.java:1070)
	at org.apache.maven.model.building.DefaultModelBuilder.readParent(DefaultModelBuilder.java:846)
	at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:337)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:292)
	... 41 more
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not find artifact org.apache:apache:pom:23 in asfsnap (http://repository.apache.org/snapshots/)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:424)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:229)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:207)
	at org.apache.maven.repository.internal.DefaultModelResolver.resolveModel(DefaultModelResolver.java:172)
	... 46 more
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.apache:apache:pom:23 in asfsnap (http://repository.apache.org/snapshots/)
	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:48)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:369)
	at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:75)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute(BasicRepositoryConnector.java:644)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get(BasicRepositoryConnector.java:262)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:499)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:401)
	... 49 more
[jbang] If you believe this a bug in jbang open issue at https://github.com/jbangdev/jbang/issues
```

As it tries to download `org.apache:apache:pom:23` but it tries with the ASF snapshot repo (where the file is not located) then it fail. That file is in maven central.

**Comment by davsclaus** [comment]
I have a launcher so you can also try it with

```
https://github.com/davsclaus/camel-bang
```

PS I am using 

❯ jbang version
0.65.1

**Comment by maxandersen** [comment]
Casing does not or at least should not matter for the shorthand names. 

I wonder if the maven central gets affected by snapshots. 

Will look when back at laptop.

**Comment by maxandersen** [comment]
okey, its the handling of mavencentral that is not working as expected. I kinda knew that could come back and bite; but I hadn't had this case of "artifacts in one repo, dependencies in another". 

It is an easy fix for you too though - until we release a new build a workaround for you is to explicitly add maven central, like:

```
//REPOS asfsnap=http://repository.apache.org/snapshots/
//REPOS mc=https://repo1.maven.org/maven2/
```

fix in jbang will be to have `mavencentral` effectively do the same thing and ignore shrinkwraps "interpretation" on how central should work ;)

**Comment by maxandersen** [comment]
this was fixed in 0.65.1.
